### ISSUE TITLE
linked module does not take directories.lib into account with 0.17.0-beta.2

### ISSUE THREAD

**Issue by henrjk** [jspm/jspm-cli#1433]
I have a module on my file system with package.json. Here an excerpt:

```
{
  "name": "anvil-connect-js",
  "version": "0.1.1",
  "main": "anvil-connect.js",
  "directories": {  <-- I have copied this here manually from the jspm section thinking it might be needed
    "lib": "src"
  },
..
  "jspm": {
    "registry": "github",
    "main": "anvil-connect",
    "format": "es6",
    "directories": {
      "lib": "src"
    },
...
```

In my app I linked the connect-js module as follows:
```
$ jspm link ~/code/follow/connect-js github:anvilresearch/connect-js
     Updating registry cache...
     Processing linked package github:anvilresearch/connect-js@0.1.1
     Downloading npm:base64-js@0.0.8
     Downloading npm:q@1.4.1
     Downloading npm:text-encoder-lite@1.0.0
     Downloading npm:tiny-emitter@1.0.2
ok   Installed npm:bows@^1.4.8 (1.4.8)
ok   Installed npm:base64-js@^0.0.8 (0.0.8)
ok   Installed npm:q@^1.4.1 (1.4.1)
ok   Installed npm:text-encoder-lite@^1.0.0 (1.0.0)
     Downloading github:vibornoff/webcrypto-shim@master
ok   Installed npm:andlog@^1.0.0 (1.0.0)
ok   Installed npm:tiny-emitter@^1.0.1 (1.0.2)

warn Unable to use the GitHub API to speed up dependency downloads due to invalid response code 404.
     Downloading github:angular/bower-angular@1.4.8
ok   Installed peer process to github:jspm/nodelibs-process@^0.2.0-alpha (0.2.0-alpha)
ok   Installed peer fs to github:jspm/nodelibs-fs@^0.2.0-alpha (0.2.0-alpha)
ok   Installed github:vibornoff/webcrypto-shim@master (master)
ok   Installed github:angular/bower-angular@^1.4.8 (1.4.8)
ok   Symlinked connect-js as github:anvilresearch/connect-js@0.1.1 (0.1.1)
     Removing package files for npm:systemjs-plugin-babel@0.0.2
     Install tree has no forks.
     
     Run this link command again or jspm install github:anvilresearch/connect-js@0.1.1 to relink changes in the package.json file.
     Run jspm install --unlink to unlink and install all original packages. Linked packages can also be uninstalled normally.
```

This looked all good to me. But when I open my app (`grunt serve`) it successfully downloads all kinds of modules but when it gets to download a module inside connect-js I see:

```
[D] server GET /jspm/jspm_packages/github/anvilresearch/connect-js@0.1.1/anvil-connect-angular.js 404 94 - 12.748 ms
```
`/jspm/jspm_packages/github/anvilresearch/connect-js@0.1.1/` is a symlink to the root of the linked project so that the **request is missing the src folder** it should be:

```
[D] server GET /jspm/jspm_packages/github/anvilresearch/connect-js@0.1.1/src/anvil-connect-angular.js 
```

Is this an issue or am I using something wrong?

The module is include by an app.js script like so:
```
import 'connect-js/anvil-connect-angular'
import 'angular-animate'
import 'angular-cookies'
import 'angular-resource'
import 'angular-route'
import 'angular-sanitize'
import 'angular-touch'
```
Here is the full list of requests I am seeing on the server:
```
[D] server GET / 200 2501 - 33.048 ms
[D] server GET /bower_components/bootstrap/dist/css/bootstrap.css 304 - - 10.654 ms
[D] server GET /jspm/jspm_packages/system.src.js 304 - - 10.726 ms
[D] server GET /jspm/jspm.browser.js 304 - - 10.213 ms
[D] server GET /jspm/jspm.js 304 - - 10.194 ms
[D] server GET /styles/main.css 304 - - 13.499 ms
[D] server GET /bower_components/bootstrap/dist/fonts/glyphicons-halflings-regular.woff 304 - - 0.794 ms
[D] server GET /scripts/app.js 200 3516 - 1.342 ms
[D] server GET /jspm/jspm_packages/npm/systemjs-plugin-babel@0.0.2.json 304 - - 0.706 ms
[D] server GET /jspm/jspm_packages/npm/systemjs-plugin-babel@0.0.2/plugin-babel.js 304 - - 1.991 ms
[D] server GET /jspm/jspm_packages/npm/systemjs-plugin-babel@0.0.2/systemjs-babel-browser.js 304 - - 0.414 ms
[D] server GET /jspm/jspm_packages/github/anvilresearch/connect-js@0.1.1.json 304 - - 0.624 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-animate@1.4.8.json 304 - - 0.589 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-cookies@1.4.8.json 304 - - 0.539 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-resource@1.4.8.json 304 - - 0.953 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-route@1.4.8.json 304 - - 1.531 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-sanitize@1.4.8.json 304 - - 2.514 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-touch@1.4.8.json 304 - - 0.559 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-animate@1.4.8/angular-animate.js 304 - - 2.478 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-cookies@1.4.8/angular-cookies.js 304 - - 3.177 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-resource@1.4.8/angular-resource.js 304 - - 3.545 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-route@1.4.8/angular-route.js 304 - - 3.637 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-sanitize@1.4.8/angular-sanitize.js 304 - - 3.937 ms
[D] server GET /jspm/jspm_packages/github/anvilresearch/connect-js@0.1.1/anvil-connect-angular.js 404 94 - 12.748 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular-touch@1.4.8/angular-touch.js 304 - - 3.943 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular@1.4.8.json 304 - - 0.392 ms
[D] server GET /jspm/jspm_packages/github/angular/bower-angular@1.4.8/angular.js 304 - - 0.846 ms
```

**Comment by henrjk** [comment]
I was just debugging this problem and jspm asked **..are you sure you want to link over it?** to which I answered Yes.
The BP didn't hit and it processed the package again. This appears to have fixed the issue as now the symlink goes to src not the root of the linked package:

```
$ ls -l jspm/jspm_packages/github/anvilresearch/
total 16
lrwxr-xr-x  1 dev  staff  29 Jan 15 09:24 connect-js@0.1.1 -> ../../../../../connect-js/src
-rw-r--r--  1 dev  staff   2 Jan 15 09:24 connect-js@0.1.1.json
```

The link over output was:
```
igelmac:connect-example-angularjs dev (use-jspm)$ node --debug-brk /Users/dev/.nvm/versions/node/v4.2.3/bin/jspm link ~/code/follow/connect-js github:anvilresearch/connect-js
Debugger listening on port 5858
github:anvilresearch/connect-js@0.1.1 is already linked, are you sure you want to link over it? [Yes]: Yes
     Updating registry cache...
     Processing linked package github:anvilresearch/connect-js@0.1.1

warn Package github:anvilresearch/connect-js@0.1.1 has no "main" entry point set in its package config.
ok   Symlinked connect-js as github:anvilresearch/connect-js@0.1.1 (0.1.1)
     Install tree has no forks.
     
     Run this link command again or jspm install github:anvilresearch/connect-js@0.1.1 to relink changes in the package.json file.
     Run jspm install --unlink to unlink and install all original packages. Linked packages can also be uninstalled normally.
``` 

I am not sure why its behaving differently now. Perhaps I added the package.json root level directories/lib entry after my initial link command.
``` 
package.json: 
{
  "directories": {  <-- I have copied this here manually from the jspm section thinking it might be needed
    "lib": "src"
  },
...
```

I leave this open to clarify whether the directories entry with the "jspm" section is also expected to work.

**Comment by guybedford** [comment]
Thanks, yes this was a bug. Fixed in https://github.com/jspm/jspm-cli/commit/33833bba65062edefc466ff45fad0a91478e6452 pending release.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.5.
